vms/platformvm/state/state_test.go

```diff
@@ -2088,3 +2088,257 @@ func TestSubnetOnlyValidatorAfterLegacyRemoval(t *testing.T) {
+		},
+		{
+			name: "initial stakers in same subnet",
+			initial: []*Staker{
+				{
+					TxID:      ids.GenerateTestID(),
+					SubnetID:  subnetID1,
+					NodeID:    ids.GenerateTestNodeID(),
+					PublicKey: pk,
+					Weight:    1,
+					StartTime: time.Now(),
+				},
+				{
+					TxID:      ids.GenerateTestID(),
+					SubnetID:  subnetID1,
+					NodeID:    ids.GenerateTestNodeID(),
+					PublicKey: otherPK,
+					Weight:    1,
+					StartTime: time.Now(),
+				},
+			},
+		},
+		{
+			name: "initial stakers in different subnets",
+			initial: []*Staker{
+				{
+					TxID:      ids.GenerateTestID(),
+					SubnetID:  subnetID1,
+					NodeID:    ids.GenerateTestNodeID(),
+					PublicKey: pk,
+					Weight:    1,
+					StartTime: time.Now(),
+				},
+				{
+					TxID:      ids.GenerateTestID(),
+					SubnetID:  subnetID2,
+					NodeID:    ids.GenerateTestNodeID(),
+					PublicKey: otherPK,
+					Weight:    1,
+					StartTime: time.Now(),
+				},
+			},
+		},
+		{
+			name: "sovs only in same subnet",
+			sovs: []SubnetOnlyValidator{
+				{
+					ValidationID: ids.GenerateTestID(),
+					SubnetID:     subnetID1,
+					NodeID:       ids.GenerateTestNodeID(),
+					StartTime:    uint64(time.Now().Unix()),
+					PublicKey:    pkBytes,
+					Weight:       1,
+				},
+				{
+					ValidationID: ids.GenerateTestID(),
+					SubnetID:     subnetID1,
+					NodeID:       ids.GenerateTestNodeID(),
+					PublicKey:    otherPKBytes,
+					StartTime:    uint64(time.Now().Unix()),
+					Weight:       1,
+				},
+			},
+		},
+		{
+			name: "sovs only in different subnets",
+			sovs: []SubnetOnlyValidator{
+				{
+					ValidationID: ids.GenerateTestID(),
+					SubnetID:     subnetID1,
+					NodeID:       ids.GenerateTestNodeID(),
+					StartTime:    uint64(time.Now().Unix()),
+					PublicKey:    pkBytes,
+					Weight:       1,
+				},
+				{
+					ValidationID: ids.GenerateTestID(),
+					SubnetID:     subnetID2,
+					NodeID:       ids.GenerateTestNodeID(),
+					PublicKey:    otherPKBytes,
+					StartTime:    uint64(time.Now().Unix()),
+					Weight:       1,
+				},
+			},
+		},
+		{
+			name: "initial stakers and sovs mixed",
+			initial: []*Staker{
+				{
+					TxID:      ids.GenerateTestID(),
+					SubnetID:  subnetID1,
+					NodeID:    ids.GenerateTestNodeID(),
+					PublicKey: pk,
+					Weight:    uint64(time.Now().Unix()),
+					StartTime: time.Now(),
+				},
+				{
+					TxID:      ids.GenerateTestID(),
+					SubnetID:  subnetID2,
+					NodeID:    ids.GenerateTestNodeID(),
+					PublicKey: pk,
+					Weight:    1,
+					StartTime: time.Now(),
+				},
+				{
+					TxID:      ids.GenerateTestID(),
+					SubnetID:  subnetID1,
+					NodeID:    ids.GenerateTestNodeID(),
+					PublicKey: otherPK,
+					Weight:    0,
+					StartTime: time.Now(),
+				},
+			},
+			sovs: []SubnetOnlyValidator{
+				{
+					ValidationID: ids.GenerateTestID(),
+					SubnetID:     subnetID1,
+					NodeID:       ids.GenerateTestNodeID(),
+					StartTime:    uint64(time.Now().Unix()),
```

Comment:
could we define a single `now := time.Now()` to avoid the possibility of different return values for time.Now() for some unlucky test runs?

Reply:
changed